stdlib/zipfile/__init__.pyi

```diff
@@ -106,24 +120,50 @@ class ZipFile:
     compresslevel: int | None  # undocumented
     mode: _ZipFileMode  # undocumented
     pwd: bytes | None  # undocumented
+    # metadata_encoding is new in 3.11
     if sys.version_info >= (3, 11):
         @overload
         def __init__(
             self,
             file: StrPath | IO[bytes],
+            mode: _ZipFileMode = "r",
+            compression: int = 0,
+            allowZip64: bool = True,
+            compresslevel: int | None = None,
+            *,
+            strict_timestamps: bool = True,
+            metadata_encoding: str | None = None,
+        ) -> None: ...
```

Comment:
Aren't all following overloads shadowed by this one?

Reply:
No - for example, `IO[bytes]` is broader than `_ZipReadable`, which will lead to the second overload getting chosen if you just an object that is readable.

Reply:
As for `_ZipFileMode`, if `file` doesn't match we'll end up skipping over the overload.

Reply:
> `IO[bytes]` is broader than `_ZipReadable`

But that's exactly the problem. It's my understanding that overloads are processed in order (although [the typing spec](https://typing.readthedocs.io/en/latest/spec/overload.html) doesn't mention that), so `IO[bytes]` always matches before `_ZipReadable` can match.

Reply:
How is that a problem though? `_ZipReadable` will get matched if `IO[bytes]` isn't fully fulfilled, or am I misunderstanding something?